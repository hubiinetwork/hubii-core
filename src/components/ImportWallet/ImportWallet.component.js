import * as React from 'react';
import { Icon } from 'antd';
import {
  Between,
  CreateButton,
  LeftArrow,
  Flex,
  Coins,
  CoinButton,
  Image,
  Center
} from './ImportWallet.style';

class ImportWallet extends React.Component {
  onChange = e => {
    this.props.changeSelectedWallet &&
      this.props.changeSelectedWallet(e.target.value);
  };
  onGoBack = e => {
    this.props.onGoBack && this.props.onGoBack();
  };
  render() {
    return (
      <div>
        <Between>
          <Flex>
            <LeftArrow type="arrow-left" onClick={this.onGoBack} />
            <span>Importing Wallet</span>
          </Flex>
          <div>
            <CreateButton>
              <Icon type="plus" />Create new wallet
            </CreateButton>
          </div>
        </Between>
        <Coins onChange={this.onChange}>
          {this.props.wallets.map((wallet, index) => (
            <CoinButton value={wallet.value} key={index}>
              <Center>
                <Image src={wallet.src} />
              </Center>
            </CoinButton>
          ))}
        </Coins>
      </div>
    );
  }
}
export default ImportWallet;
