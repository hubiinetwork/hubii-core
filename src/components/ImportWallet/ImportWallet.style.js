import styled from 'styled-components';
import Button from '../ui/Button';
import { Icon, Radio } from 'antd';

export const Between = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
`;

export const CreateButton = styled(Button)`
  background-color: transparent;
  color: ${({ theme }) => theme.palette.light};
  &:hover {
    background-color: ${({ theme }) => theme.palette.info2};
    color: ${({ theme }) => theme.palette.light};
    border-color: ${({ theme }) => theme.palette.light};
  }
`;

export const LeftArrow = styled(Icon)`
  font-size: 20px;
  margin-right: 7px;
  cursor: pointer;
`;

export const Coins = styled(Radio.Group)`
  margin-top: 32px;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  .ant-radio-button-wrapper-checked {
    background-color: ${({ theme }) => theme.palette.info};
    border: 1px solid ${({ theme }) => theme.palette.info};
  }
`;
const RadioButton = Radio.Button;
export const CoinButton = styled(RadioButton)`
  margin-top: 15px;
  padding-left: 25px;
  padding-right: 25px;
  padding-top: 15px;
  padding-bottom: 40px;
  background-color: transparent;
  min-width: 230px;
  min-height: 60px;
  border-radius: 3px;
`;

export const Flex = styled.div`
  display: flex;
  color: ${({ theme }) => theme.palette.light};
`;

export const Image = styled.img`
  width: 90px;
  height: 30px;
`;

export const Center = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
`;
export const SpanText = styled.span`
  font-size: 16px;
  font-weight: 500;
  line-height: 19px;
`;
