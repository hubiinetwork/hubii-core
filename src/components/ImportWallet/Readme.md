Variants of ContactList:

```js
<div style={{ backgroundColor: '#2f4d5c', padding: 25 }}>
  <ImportWallet />
</div>
```
