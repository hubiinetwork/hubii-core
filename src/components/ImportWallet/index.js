export { default } from './ImportWallet.component';
