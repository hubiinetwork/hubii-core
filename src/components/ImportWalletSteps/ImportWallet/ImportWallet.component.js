import * as React from 'react';
import PropTypes from 'prop-types';
import {
  Coins,
  Image,
  Center,
  ButtonDiv,
  CoinButton,
  StyledSpan,
  StyledButton,
} from './ImportWallet.style';

class ImportWallet extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      walletType: '',
    };
    this.onChange = this.onChange.bind(this);
    this.handleNext = this.handleNext.bind(this);
  }

  onChange(e) {
    this.setState({ walletType: e.target.value });
  }

  handleNext(e) {
    const { handleNext } = this.props;
    const { walletType } = this.state;
    e.preventDefault();
    handleNext({ walletType });
  }

  render() {
    const { walletType } = this.state;
    return (
      <div>
        <Coins onChange={this.onChange}>
          {this.props.wallets.map((wallet) => (
            <CoinButton value={wallet.name} key={wallet.name}>
              <Center>
                <Image src={wallet.src} />
              </Center>
            </CoinButton>
          ))}
        </Coins>
        <ButtonDiv>
          <StyledButton type={'primary'} disabled={walletType === ''} onClick={this.handleNext}>
            <StyledSpan>Next</StyledSpan>
          </StyledButton>
        </ButtonDiv>
      </div>
    );
  }
}

ImportWallet.propTypes = {
  /**
   * Array of contacts whose list is to be shown.
   */
  wallets: PropTypes.array.isRequired,
    /**
   * Function to be executed when next button is pressed
   */
  handleNext: PropTypes.func,

};

export default ImportWallet;
