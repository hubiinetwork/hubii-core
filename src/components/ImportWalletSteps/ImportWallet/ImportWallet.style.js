import { Radio } from 'antd';
import styled from 'styled-components';
import Button from '../../ui/Button';

export const Coins = styled(Radio.Group)`
  margin-top: 40px;
  margin-bottom: 50px;
  padding-left: 27px;
  padding-right: 27px;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  .ant-radio-button-wrapper-checked {
    background-color: ${({ theme }) => theme.palette.info};
    border: 1px solid ${({ theme }) => theme.palette.info};
  }
`;
const RadioButton = Radio.Button;

export const CoinButton = styled(RadioButton)`
  padding: 0.5rem;
  display: flex;
  justify-content: center;
  align-items: center;
  margin-top: 15px;
  padding-top: 8px;
  background-color: transparent;
  width: 227px;
  height: 48px;
  border-radius: 3px !important;
`;

export const Flex = styled.div`
  display: flex;
  color: ${({ theme }) => theme.palette.light};
`;

export const Image = styled.img`
  width: 90px;
  height: 30px;
`;

export const OptionText = styled.h3`
  color: white;
  vertical-align: center;
`;

export const Center = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
`;
export const SpanText = styled.span`
  font-size: 16px;
  font-weight: 500;
  line-height: 19px;
`;
export const ButtonDiv = styled.div`
  display: flex;
  justify-content: center;
  margin-bottom: 26px;
  margin-top: 1rem;
`;
export const StyledButton = styled(Button)`
  margin-top: 80px;
  background-color: ${({ disabled: white }) =>
    white && 'transparent !important'};
  font-size: 12px;
  font-weight: 500;
  border-width: 2px;
  height: 40px;
  width: 162px;
  border: ${({ disabled: white, theme }) =>
    white && `2px solid ${theme.palette.secondary4} !important`};
  min-width: ${({ current: width }) => (width === 0 ? '260px' : '260px')};
  color: ${({ disabled: white, theme }) =>
    white
      ? `${theme.palette.secondary4} !important`
      : `${theme.palette.light} !important`};
  &:hover {
    background-color: ${({ disabled: white }) =>
      white && 'transparent !important'};
    border: ${({ disabled: white, theme }) =>
      white && `2px solid ${theme.palette.secondary4} !important`};
  }
`;
export const StyledSpan = styled.span`
  font-size: 12px;
  font-weight: 500;
  line-height: 14px;
  text-align: center;
`;
