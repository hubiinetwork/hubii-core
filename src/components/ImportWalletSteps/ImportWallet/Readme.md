Variants of ImportWallet:

```js
const walletData = [
  {
    src:
      'https://www.ledger.fr/wp-content/uploads/2017/09/Ledger_logo_footer@2x.png',
    value: 'ledger'
  },
  {
    src: 'https://new.consensys.net/wp-content/uploads/2018/01/Metamask.png',
    value: 'metamask'
  },
  {
    src:
      'https://cdn-images-1.medium.com/max/1600/1*u3_I95cOdCBd3gBJrBd2Aw.png',
    value: 'parity'
  },
  {
    src: 'https://pbs.twimg.com/media/Cxy4iJVXcAMJr9y.png',
    value: 'digitalBitbox'
  },
  {
    src:
      'https://www.ledger.fr/wp-content/uploads/2017/09/Ledger_logo_footer@2x.png',
    value: 'ledger1'
  },
  {
    src: 'https://new.consensys.net/wp-content/uploads/2018/01/Metamask.png',
    value: 'metamask1'
  },
  {
    src:
      'https://cdn-images-1.medium.com/max/1600/1*u3_I95cOdCBd3gBJrBd2Aw.png',
    value: 'parity1'
  }
];
<div
  style={{
    backgroundColor: '#2f4d5c',
    padding: 25,
    width: 576,
    height: 576
  }}
>
  <ImportWallet wallets={walletData} />
</div>;
```
