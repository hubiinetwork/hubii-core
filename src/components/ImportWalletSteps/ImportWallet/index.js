import * as React from 'react';
import PropTypes from 'prop-types';
import { injectIntl } from 'react-intl';

import Text from 'components/ui/Text';

import {
  OptionsWrapper,
  Image,
  Center,
  Option,
  StyledButton,
  OptionText,
  Header,
  Wrapper,
} from './style';

class ImportWallet extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      walletType: '',
    };
    this.onChange = this.onChange.bind(this);
    this.handleNext = this.handleNext.bind(this);
  }

  onChange(e) {
    this.setState({ walletType: e.target.value });
  }

  handleNext(e) {
    const { handleNext } = this.props;
    const { walletType } = this.state;
    e.preventDefault();
    handleNext({ walletType });
  }

  render() {
    const { walletType } = this.state;
    const { formatMessage } = this.props.intl;
    return (
      <Wrapper>
        <Header>{formatMessage({ id: 'store_wallet_question' })}</Header>
        <OptionsWrapper onChange={this.onChange}>
          {this.props.wallets.map((wallet) => (
            <Option value={wallet.name} key={wallet.name}>
              <Center>
                {
                  wallet.src ? <Image src={wallet.src} /> : <OptionText>{wallet.name}</OptionText>
                }
              </Center>
            </Option>
          ))}
        </OptionsWrapper>
        <StyledButton type={'primary'} disabled={walletType === ''} onClick={this.handleNext}>
          <Text>{formatMessage({ id: 'next' })}</Text>
        </StyledButton>
      </Wrapper>
    );
  }
}

ImportWallet.propTypes = {
  /**
   * Array of contacts whose list is to be shown.
   */
  wallets: PropTypes.array.isRequired,
  /**
   * Function to be executed when next button is pressed
   */
  handleNext: PropTypes.func,
  intl: PropTypes.object.isRequired,
};

export default injectIntl(ImportWallet);
