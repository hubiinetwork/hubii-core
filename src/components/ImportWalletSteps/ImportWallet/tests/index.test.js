import React from 'react';
import { shallow } from 'enzyme';
import { intl } from 'jest/__mocks__/react-intl';
import ImportWallet from '../index';

const walletData = [
  {
    src:
        'https://www.ledger.fr/wp-content/uploads/2017/09/Ledger_logo_footer@2x.png',
    name: 'ledger',
  },
  {
    name: 'Private key',
  },
];

describe('<ImportWallet />', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<ImportWallet wallets={walletData} intl={intl} />);
    expect(wrapper).toMatchSnapshot();
  });
});
