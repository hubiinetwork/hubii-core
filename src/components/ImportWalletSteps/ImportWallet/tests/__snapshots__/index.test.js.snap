// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImportWallet /> should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ImportWallet
    intl={
        Object {
            "formatMessage": [Function],
          }
    }
    wallets={
        Array [
            Object {
              "name": "ledger",
              "src": "https://www.ledger.fr/wp-content/uploads/2017/09/Ledger_logo_footer@2x.png",
            },
            Object {
              "name": "Private key",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <style__Header>
          store_wallet_question
</style__Header>,
        <style__OptionsWrapper
          onChange={[Function]}
>
          <style__Option
                    value="ledger"
          >
                    <style__Center>
                              <style__Image
                                        src="https://www.ledger.fr/wp-content/uploads/2017/09/Ledger_logo_footer@2x.png"
                              />
                    </style__Center>
          </style__Option>
          <style__Option
                    value="Private key"
          >
                    <style__Center>
                              <style__OptionText>
                                        Private key
                              </style__OptionText>
                    </style__Center>
          </style__Option>
</style__OptionsWrapper>,
        <style__StyledButton
          disabled={true}
          onClick={[Function]}
          type="primary"
>
          <Text>
                    next
          </Text>
</style__StyledButton>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "store_wallet_question",
        },
        "ref": null,
        "rendered": "store_wallet_question",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <style__Option
              value="ledger"
>
              <style__Center>
                            <style__Image
                                          src="https://www.ledger.fr/wp-content/uploads/2017/09/Ledger_logo_footer@2x.png"
                            />
              </style__Center>
</style__Option>,
            <style__Option
              value="Private key"
>
              <style__Center>
                            <style__OptionText>
                                          Private key
                            </style__OptionText>
              </style__Center>
</style__Option>,
          ],
          "onChange": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "ledger",
            "nodeType": "class",
            "props": Object {
              "children": <style__Center>
                <style__Image
                                src="https://www.ledger.fr/wp-content/uploads/2017/09/Ledger_logo_footer@2x.png"
                />
</style__Center>,
              "value": "ledger",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <style__Image
                  src="https://www.ledger.fr/wp-content/uploads/2017/09/Ledger_logo_footer@2x.png"
/>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "src": "https://www.ledger.fr/wp-content/uploads/2017/09/Ledger_logo_footer@2x.png",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "Private key",
            "nodeType": "class",
            "props": Object {
              "children": <style__Center>
                <style__OptionText>
                                Private key
                </style__OptionText>
</style__Center>,
              "value": "Private key",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <style__OptionText>
                  Private key
</style__OptionText>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Private key",
                },
                "ref": null,
                "rendered": "Private key",
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Text>
            next
</Text>,
          "disabled": true,
          "onClick": [Function],
          "type": "primary",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "next",
          },
          "ref": null,
          "rendered": "next",
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <style__Header>
            store_wallet_question
</style__Header>,
          <style__OptionsWrapper
            onChange={[Function]}
>
            <style__Option
                        value="ledger"
            >
                        <style__Center>
                                    <style__Image
                                                src="https://www.ledger.fr/wp-content/uploads/2017/09/Ledger_logo_footer@2x.png"
                                    />
                        </style__Center>
            </style__Option>
            <style__Option
                        value="Private key"
            >
                        <style__Center>
                                    <style__OptionText>
                                                Private key
                                    </style__OptionText>
                        </style__Center>
            </style__Option>
</style__OptionsWrapper>,
          <style__StyledButton
            disabled={true}
            onClick={[Function]}
            type="primary"
>
            <Text>
                        next
            </Text>
</style__StyledButton>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "store_wallet_question",
          },
          "ref": null,
          "rendered": "store_wallet_question",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <style__Option
                value="ledger"
>
                <style__Center>
                                <style__Image
                                                src="https://www.ledger.fr/wp-content/uploads/2017/09/Ledger_logo_footer@2x.png"
                                />
                </style__Center>
</style__Option>,
              <style__Option
                value="Private key"
>
                <style__Center>
                                <style__OptionText>
                                                Private key
                                </style__OptionText>
                </style__Center>
</style__Option>,
            ],
            "onChange": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "ledger",
              "nodeType": "class",
              "props": Object {
                "children": <style__Center>
                  <style__Image
                                    src="https://www.ledger.fr/wp-content/uploads/2017/09/Ledger_logo_footer@2x.png"
                  />
</style__Center>,
                "value": "ledger",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <style__Image
                    src="https://www.ledger.fr/wp-content/uploads/2017/09/Ledger_logo_footer@2x.png"
/>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "src": "https://www.ledger.fr/wp-content/uploads/2017/09/Ledger_logo_footer@2x.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "Private key",
              "nodeType": "class",
              "props": Object {
                "children": <style__Center>
                  <style__OptionText>
                                    Private key
                  </style__OptionText>
</style__Center>,
                "value": "Private key",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <style__OptionText>
                    Private key
</style__OptionText>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Private key",
                  },
                  "ref": null,
                  "rendered": "Private key",
                  "type": [Function],
                },
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Text>
              next
</Text>,
            "disabled": true,
            "onClick": [Function],
            "type": "primary",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "next",
            },
            "ref": null,
            "rendered": "next",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
